docs/ui/CopyButton.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Button } from '@marigold/components';
+
+interface CopyProps {
+  codeString: string;
+}
+
+export const CopyButton = ({ codeString }: CopyProps) => {
+  const [isCopied, setIsCopied] = React.useState(false);
```

Comment:
we should use a hook for this :)

Reply:
because we need to change the copy icon based on if it is copy or not

Reply:
We could use something like this too :)

https://www.npmjs.com/package/react-use-clipboard